plugins/janus_videoroom.c

```diff
@@ -1115,6 +1127,26 @@ static void janus_videoroom_notify_participants(janus_videoroom_participant *par
 	}
 }
 
+static void janus_videoroom_participant_joining(janus_videoroom_participant * p) {
+	/* we need to check if the room still exists, may have been destroyed already */
+	if(p->room && !p->room->destroyed && p->room->notify_joining) {
+		json_t *event = json_object();
+		json_t *user = json_object();
+		json_object_set_new(user, "id", json_integer(p->user_id));
+		if (p->display) {
+			json_object_set_new(user, "display", json_string(p->display));
+		}
+		json_object_set_new(event, "videoroom", json_string("event"));
+		json_object_set_new(event, "room", json_integer(p->room->room_id));
+		json_object_set_new(event, "joining", user);
+		janus_mutex_lock(&p->room->participants_mutex);
+		janus_videoroom_notify_participants(p, event);
+		janus_mutex_unlock(&p->room->participants_mutex);
+		json_decref(event);
+		json_decref(user);
```

Comment:
This is not needed: `json_object_set_new` means `event` now owns `user`. Dereferencing `user` at this point is likely going to cause issues and maybe crashes. This might explain why your events were not working as expected for me: the content of "joining" was not what you added, but some Janus core API stuff that Jansson probably incorrectly added due to broken pointers.

Reply:
I can confirm that removing the `json_decref(user)` line fixes the issue for me.